metabase/metabase
# Fix issues in auto apply button

Epic https://github.com/metabase/metabase/issues/29267

Fixes:
- Auto apply button flashing when parameters are set
- Preserves parameter values when auto apply setting is toggled off via the toast
- Uses correct (= last applied) parameter values when toggling auto applying on and off
- Removes icon from the toast
- Prevents showing the toast for the same dashboard

Changes:
- Moves tests originally added to `old-parameters.cy.spec.js` to a separate file. This test suite was intended for legacy parameters and not for new features.
- Moves auto apply filters logic to redux
- Adds e2e tests

How to verify:
- Open a dashboard with multiple parameters and questions
- Info -> Turn off Auto-apply filters
- Change filters and make sure that changes are applied only when Apply button is clicked


- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30613)
